examples/docker/Dockerfile

```diff
@@ -0,0 +1,11 @@
+FROM node:16-alpine
```

Comment:
If this is meant to be built for production, there's some suggestions here:

1) You should be specific on the version of node, such as `node:16.14.2-alpine`. 
This prevents possible regressions (if any occurred). There's some other arguments to be made here, so take this as you wish.

2) You should keep out dev-dependencies
Docker builds should be as slim as possible, and having all your files (including your .git, source and dist files) is not a good practice. 

3) You should switch to a non-root user, in this case node provides a `node` user.
Sandboxing!

I suggest this be switched to a more complex, but "production-hardened" dockerfile with another docker/compose file with development in mind. That way, you can have a secure but useful dockerfile for your production, but then have something a bit relaxed for development.

Maybe there's an argument to have a basic setup like this, but I think providing more than this will be super beneficial to everyone.

Reply:
1. I didn't think this would be a problem but it is better to specify the version. 
2. i think yarn install --production will solve this, 

> Yarn will not install any package listed in devDependencies if the NODE_ENV environment variable is set to production

4. for this:
```
RUN addgroup -g 1001 -S nodejs
RUN adduser -S remix -u 1001
USER remix 
```
add this code to dockerfile will be good ? 

i am not docker expert but i like use it and this my first time to contributing in project. so when i saw the examples and there no docker example i think why i don't add it. i want this example to be very simple.
i tryed to find how to write Docker for dev and Live Reloading the code but i could not find solution. all what i found is to use Volumes and did not work with me :(